qutip/core/metrics.py

```diff
@@ -105,12 +108,104 @@ def fidelity(A, B):
     return float(np.real(np.sqrt(eig_vals[eig_vals > 0]).sum()))
 
 
-def process_fidelity(U1, U2, normalize=True):
+def _process_fidelity_to_id(oper):
+    """
+    Internal function returning the process fidelity of a quantum channel
+    to the identity quantum channel.
+    Parameters
+    ----------
+    oper : :class:`qutip.Qobj`/list
+        A unitary operator, or a superoperator in supermatrix, Choi or
+        chi-matrix form, or a list of Kraus operators
+    Returns
+    -------
+    fid : float
     """
-    Calculate the process fidelity given two process operators.
+    msg = 'The process fidelity to identity is only defined for ' \
+          'dimension preserving channels.'
+    if isinstance(oper, list):  # oper is a list of Kraus operators
+        d = oper[0].shape[0]
+        if oper[0].shape[1] != d:
+            raise TypeError(msg)
```

Comment:
```suggestion
            raise ValueError(msg)
```
I suggest we change these to `ValueError`. The formal type (i.e. list or Qobj) is correct, but the value is not correct (i.e. channel has the wrong shape).

Reply:
The same comment applies to the other `TypeError` exceptions too.

Also happy for @Ericgig to chime in with whether he agrees with my suggestion or not and to leave it as is if he doesn't.

Reply:
In the rest of this file, metrics raise a `TypeError` when the shape doesn't match, for example in `fidelity` and `trace_dist`. So I'd vote to keep `TypeError` here for consistency.

Reply:
Happy with that -- we can always fix these TypeErrors in a follow up PR if we decide to.